### PR TITLE
Add API for communicating with Docker volume plugins

### PR DESCRIPTION
**Pull request by mheon**
Docker provides extensibility through a plugin system, of which several types are available. This provides an initial library API for communicating with one type of plugins, volume plugins. Volume plugins allow for an external service to create and manage a volume on Podman's behalf.

This does not integrate the plugin system into Libpod or Podman yet; that will come in subsequent pull requests.

This replaces #4548 and will begin the continuation of that work.

### CODE REVIEW

**Comment by mheon** [comment]
I'm especially interested from a review from @jwhonce on the HTTP bits.

**Comment by mheon** [comment]
@rhatdan will also be interested in this.

File: go.mod
Comment (mheon) [inline go.mod:16]: Eeek. I don't know how this got in here...

**Review by jwhonce — COMMENTED** [review]
This is using a seperate HTTP Server from current API server. That is not "bad" but it does imply as written: 
1) people looking at this code need to know this server configuration is not impacted by pkg/api/server, 
2) this is hard-wiring a unix domain socket on the server while the imported library does allow tcp connections

I also didn't catch where the socket is being created/cleaned up

**Comment by mheon** [comment]
This is not a server, it is a client. The servers are run by individual plugins; we connect to the plugins via Unix socket and speak their REST API when we want them to perform services for us. The lifecycle of their sockets is external to Podman and is the business of the plugin creator.

File: libpod/plugin/volume_api.go
Comment (jwhonce) [inline libpod/plugin/volume_api.go:91]:
```suggestion
	}
	defer resp.Body.Close()
```

File: libpod/plugin/volume_api.go
Comment (jwhonce) [inline libpod/plugin/volume_api.go:293]:
```suggestion
	}
	defer resp.Body.Close()
```

**Review by jwhonce — CHANGES_REQUESTED** [review]

File: libpod/plugin/volume_api.go
Comment (TomSweeneyRedHat) [inline libpod/plugin/volume_api.go:109]:
```suggestion
		return errors.Wrapf(err, "error unmarshalling plugin %s activation response", newPlugin.Name)
```

File: libpod/plugin/volume_api.go
Comment (TomSweeneyRedHat) [inline libpod/plugin/volume_api.go:299]:
```suggestion
		return nil, errors.Wrapf(err, "error unmarshalling volume plugin %s list response", p.Name)
```

File: libpod/plugin/volume_api.go
Comment (TomSweeneyRedHat) [inline libpod/plugin/volume_api.go:334]:
```suggestion
		return nil, errors.Wrapf(err, "error unmarshalling volume plugin %s get response", p.Name)
```

File: libpod/plugin/volume_api.go
Comment (TomSweeneyRedHat) [inline libpod/plugin/volume_api.go:390]:
```suggestion
		return "", errors.Wrapf(err, "error unmarshalling volume plugin %s path response", p.Name)
```

File: libpod/plugin/volume_api.go
Comment (TomSweeneyRedHat) [inline libpod/plugin/volume_api.go:427]:
```suggestion
		return "", errors.Wrapf(err, "error unmarshalling volume plugin %s path response", p.Name)
```

**Comment by umohnani8** [comment]
code LGTM once the suggestions are addressed.

**Comment by mheon** [comment]
All comments addressed

**Comment by TomSweeneyRedHat** [comment]
@mheon looks like a gofmt issue with `/libpod/plugin/volume_api.go`

**Comment by mheon** [comment]
Force-pushed with fix

**Comment by mheon** [comment]
@containers/podman-maintainers This is ready

**Comment by rhatdan** [comment]
@mheon I don't think this should go in until we do 3.0.

**Comment by mheon** [comment]
Sure, probably a good call.

**Comment by mheon** [comment]
3.0 is landed, this is ready now @containers/podman-maintainers

**Comment by umohnani8** [comment]
LGTM @baude @rhatdan PTAL

**Pull request merged**
